content/en/docs/tasks/security/authorization/authz-ingress/index.md

```diff
@@ -31,143 +26,390 @@ through the Istio ingress gateway with this command:
     $ kubectl apply -f <(istioctl kube-inject -f @samples/httpbin/httpbin-gateway.yaml@) -n foo
     {{< /text >}}
 
-* See [Source IP for Services with `Type=NodePort`](https://kubernetes.io/docs/tutorials/services/source-ip/#source-ip-for-services-with-type-nodeport)
-for more information. Update the ingress gateway to set `externalTrafficPolicy: local` to preserve the
-original client source IP on the ingress gateway using the following command:
+* Turn on RBAC debugging in Envoy for the ingress gateway:
 
     {{< text bash >}}
-    $ kubectl patch svc istio-ingressgateway -n istio-system -p '{"spec":{"externalTrafficPolicy":"Local"}}'
+    $ kubectl get pods -n istio-system | grep ingress | awk '{print $1}' | while read -r pod; do istioctl proxy-config log "$pod" -n istio-system --level rbac:debug; done
```

Comment:
This seems like a pretty crazy way to access the ingress gateway. Why not something like this?
```suggestion
    $ istioctl proxy-config log $(kubectl get pod -l istio=ingressgateway -n istio-system) -n istio-system --level rbac:debug
```

Reply:
You have to run istioctl proxy-config for each ingress gateway pod, so you can't escape doing a loop if there is more than one ingress gateway.  Also, you need to grab just the pod name output from get pods with cut or awk. I used awk.

Reply:
I made the other changes, and the remaining comments are related to this kind of loop.  Let me know if my answer is acceptable here, or if you still want me to rework it.

Reply:
I see. The grep and awk still seem a little ugly. Would something like this work?
```
kubectl get pods -n istio-system -l istio=ingressgateway -o name | while ...
```

Reply:
That is close, but it prints the pods in this format `pod/istio-ingressgateway-99b56479c-46zt2` and istioctl doesn't accept the `pod/` part.  We could get rid of the grep that way, but the awk (or sed) will end up being a little more complex since a substitution will be necessary to remove `pod/`.  What do you think?

Reply:
I don't know. Does this look any better?
```
kubectl get pods -n istio-system -l istio=ingressgateway -o name | sed 's|pod/||' | while ...
```
I'll leave it up to you.

Reply:
Sure, that seems reasonable.